src/Servers/Kestrel/Transport.Quic/src/Internal/QuicStreamContext.cs

```diff
@@ -129,7 +130,7 @@ public void Initialize(QuicStream stream)
 
         public override string ConnectionId
         {
-            get => _connectionId ??= $"{_connection.ConnectionId}:{StreamId}";
+            get => _connectionId ??= StringUtilities.ConcatAsHexSuffix(_connection.ConnectionId, ':', (uint)StreamId);
```

Comment:
Can we add a test verifying that the TraceIdentifier that's logged start's with the MultiplexedConnectionContext.ConnectionId and ends with the StreamId as hex?